wfsim/strax_interface.py

```diff
@@ -353,19 +357,20 @@ class FaxSimulatorPlugin(strax.Plugin):
     def setup(self):
         c = self.config
         c.update(get_resource(c['fax_config'], fmt='json'))
-        # Update gains to the nT defaults
-        self.to_pe = get_to_pe(self.run_id, c['gain_model'],
+        overrides = self.config['fax_config_override']
+        if overrides is not None:
+            c.update(overrides)
+
+        cmt_run_id = c['cmt_run_id'] if c['cmt_run_id'] != '0' else self.run_id
+        # Update gains depending on gain model
+        self.to_pe = get_to_pe(cmt_run_id, c['gain_model'],
                               c['channel_map']['tpc'][1]+1)
```

Comment:
If I understand correctly how the `get_to_pe` function works, I think there are two potential errors in here that we want to avoid:
1. If `cmt_run_id!=0` and `self.run_id` is not an integer (and it probably won't), the run will crash because CMT expects integers.
2. If `cmt_run_id==0` and `self.run_id` is an integer, the simulation file has probably the wrong name, and the user mostly likely doesn't want to query the gains from run `self.run_id`.

Can we just change this logic to something like ... ?
 ```
cmt_run_id = c['cmt_run_id']
if cmt_run_id!=0:
   self.to_pe = get_to_pe(cmt_run_id, c['gain_model'],...)
 ```
 
 and otherwise nothing is done, i.e. the default `gain_model` loaded with the context remains.

Reply:
I think it is fine, `get_to_pe` and `get_corrections_config` in general takes run id like `'012605'`. I guess I can add an assertion just to be safe.
`assert isinstance(cmt_run_id, str), "cmt_run_id should be a string not %s" % type(cmt_run_id).__name__`

Reply:
I am not sure this would solve the issue. Let me drop some examples of the cases 1 and 2 I wrote above:
1. `cmt_run_id!=0` & `self.run_id=mc_run_0001` -> CMT crashes because this is a string, but needs to be converted into an int at some point.
2. `cmt_run_id!=0` & many int MC run_ids (many jobs running at the same time for a single simulation!): `self.run_id=00001`, `self.run_id=00002`, ..., `self.run_id=10000` -> In this case we don't want to call the CMT for all these different runs, so `self.run_id` must play zero role in the CMT call.

That's why I think that the only option is to have always some working defaults (as you just set with Joran), and just call CMT for `cmt_run_id!=0`.

Does this make sense?